src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -342,23 +338,10 @@ double SolverInterfaceImpl::initialize()
   // result of _couplingScheme->getNextTimestepMaxLength() can change when calling _couplingScheme->initialize(...) and first participant method is used for setting the time window size.
   _couplingScheme->initialize(time, timeWindow);
 
-  if (_couplingScheme->hasDataBeenReceived()) {
-    performDataActions({action::Action::READ_MAPPING_PRIOR}, 0.0);
-    mapReadData();
-    performDataActions({action::Action::READ_MAPPING_POST}, 0.0);
-  }
-
-  for (auto &context : _accessor->readDataContexts()) {
-    context.moveToNextWindow();
-  }
-
-  _couplingScheme->receiveResultOfFirstAdvance();
-
-  if (_couplingScheme->hasDataBeenReceived()) {
-    performDataActions({action::Action::READ_MAPPING_PRIOR}, 0.0);
-    mapReadData();
-    performDataActions({action::Action::READ_MAPPING_POST}, 0.0);
-  }
+  performDataActions({action::Action::READ_MAPPING_PRIOR}, 0.0);
+  std::vector<double> receiveTimes{time::Storage::WINDOW_START, time::Storage::WINDOW_END};
+  mapReadData(receiveTimes);
+  performDataActions({action::Action::READ_MAPPING_POST}, 0.0);
```

Comment:
Possible reason for failing tests: We now always perform the mapping to be able to also store zero initial data in the waveform.

Reply:
Mmh, not sure why this would lead to failing tests.
But why do we need to map zero initial data in the first place?
And in any case, this sounds like sth that should be documented directly in the code.

Reply:
I found the problem with the last failing test. Will write a comment at the right place. The problem was not here.